opacus/tests/privacy_engine_test.py

```diff
@@ -550,10 +550,13 @@ def test_parameters_match(self):
             )
 
     @given(
-        noise_scheduler=st.sampled_from([None, StepNoise]),
+        has_noise_scheduler=st.booleans(),
+        has_grad_clip_scheduler=st.booleans(),
     )
     @settings(deadline=None)
-    def test_checkpoints(self, noise_scheduler: Optional[Type[StepNoise]]):
+    def test_checkpoints(
```

Comment:
Could you break this function into multiple smaller functions to make flake8 happy?

Reply:
Yeah, looks like that function was right at the limit for complexity and I crossed it :D Will refactor